fccodegen/_generator.py

```diff
@@ -1,26 +1,180 @@
+
+        # Remove quantities (units)
+        eq = eq.subs({q: 1 for q in eq.atoms(Quantity)}, simultaneous=True)
+
+        eqs.append(eq)
+
+    return eqs
+
+
+def get_unique_names(graph):
+    """
+    Creates unique names for all symbols in a cellml model graph.
+    """
+
+    # Create a symbol => name mapping, and a reverse name => symbol mapping
+    symbols = {}
+    reverse = {}
+
+    def uname(name):
+        """ Add an increasing number to a name until it's unique """
+        root = name + '_'
+        i = 0
+        while name in reverse:
+            i += 1
+            name = root + str(i)
+        return name
+
+    for v in graph:
+        if isinstance(v, sp.Derivative):
+            continue
+
+        # Try simple name
+        parts = v.name.split('$')
+        assert len(parts) == 2
+        name = parts[-1]
+
+        # If already taken, rename _both_ variables using component name
+        if name in reverse:
+
+            # Get existing variable
+            other = reverse[name]
+
+            # Check it hasn't been renamed already
+            if symbols[other] == name:
+                oparts = other.name.split('$')
+                assert len(oparts) == 2
+                oname = uname(oparts[0] + '_' + oparts[1])
+                symbols[other] = oname
+                reverse[oname] = other
+
+            # Get new name for v
+            name = uname(parts[0] + '_' + parts[1])
+
+        # Store symbol name
+        symbols[v] = name
+        reverse[name] = v
+
+    return symbols
+
+
+def get_state_list(graph):
+    """
+    Returns a list of state variables found in the given model graph.
+    """
+    states = []
+    for v in graph:
+        if isinstance(v, sp.Derivative):
+            states.append(v.args[0])
+    return states
+
+
 def create_weblab_model(model, path):
     """
     Takes a :class:`cellmlmanip.Model`, generates a ``.pyx`` model for use
     with the Web Lab, and stores it at ``path``.
     """
+    # Get equation graph
+    graph = model.get_equation_graph()
+
+    # Get ordered equations
+    eqs = get_equations(graph)
+
+    # Get unique names for all symbols
+    unames = get_unique_names(graph)
+
+    # Symbol naming function
+    def symbol_name(symbol):
+        return 'var_' + unames[symbol]
+
+    # Derivative naming function
+    def derivative_name(deriv):
+        var = deriv.args[0] if isinstance(deriv, sp.Derivative) else deriv
+        return 'd_dt_' + unames[var]
+
+    # Create expression printer
+    p = cg.WebLabPrinter(symbol_name, derivative_name)
+
+    # Create a list of state variable indices, variables, and derivative vars
+    states = []
+    for i, state in enumerate(get_state_list(graph)):
+        states.append((i, symbol_name(state), derivative_name(state)))
+
+    # Create the RHS equations
+    rhs_equations = []
+    for eq in eqs:
+        lhs, rhs = eq.lhs, eq.rhs
+        rhs_equations.append(p.doprint(lhs) + ' = ' + p.doprint(rhs))
```

Comment:
I'm kind of surprised that `p.doprint` can't take `eq` as an argument and do the right thing!

Reply:
It can but that prints equality "==" rather than assignment "="!

Reply:
Alas!